app/models/download.rb

```diff
@@ -51,10 +51,15 @@ def file_type_from_mime(mime)
     end
   end
 
-  # The base name of the file (eg, "War and Peace")
+  # The base name of the file (e.g., "War_and_Peace")
   def file_name
     name = clean(work.title)
-    name += " Work #{work.id}" if name.length < 3
+    extender = "Work_#{work.id}"
+    if name.length.zero?
+      name = extender
+    elsif name.length < 3
+      name += "_#{extender}"
+    end
```

Comment:
My JS-addled immediately thought of "simplifying" that using `join`:
```rb
name = [name, "Work_#{work.id}"].compact_blank.join("_") if name.length < 3
```
But not exactly sure this is indeed better.